enterprisemediawiki/Meza1
# Enable install on CentOS 7

Our first real server may be RHEL 7 instead of RHEL 6. All previous assumptions were for 6. We need to make it work on CentOS 7.

Since `setup.sh` downloads the Meza1 repository without git, and this change requires that the `centos7` branch be used, that branch is currently hard-coded into line 20 of `setup.sh`. **Prior to merge it must be changed back to master.**

## List of changes:

* Move firewall changes from `setup.sh` to `apache.sh` where they belong
* Make CentOS 7 use `firewall-cmd` (aka "firewalld") instead of `iptables` since iptables has been removed
* Files `ifcfg-eth0` and `ifcfg-eth` do not exist in CentOS 7 and are replaced by `ifcfg-enp0s3` and `ifcfg-enp0s8`, respectively.
  * Add default `ifcfg-enp0s8` file to Meza1
  * Apply `ifcfg-eth0` modifications to `ifcfg-enp0s3`
* Make `mysql.sh` get the correct version of Mysql Community Release
* Change from using RPMForge to EPEL for libmcrypt-devel since RPMForge evidently doesn't have it for CentOS 7
* Removed PHP imap support
  * We weren't using it
  * MediaWiki has no use for it
  * It has caused issues on getting an installable version in the past
  * It's not part of CentOS 7 repositories we were using
  * It is part of EPEL, which we're using starting in this PR, so we could get it from there...but what's the point if we're not using IMAP anyway?
* Changed early usage of `wget` to `curl` since `curl` is preinstalled on CentOS
  * Fewer commands to get install started
  * Later usage of `wget` okay since it's installed by `yums.sh`
* Removed some unnecessary commented-out lines from `setup.sh`

## How to test this branch:

This branch includes changes to `setup.sh`, so it is required that you have a totally fresh install of CentOS. I recommend doing a new CentOS install and taking a snapshot as soon as it is complete, before running any Meza1 stuff.

Additionally, this must be tested on CentOS 7 (obviously) as well as CentOS 6. This must be done to make sure the version 7 changes don't break version 6. Additionally, to be safe we should also test on CentOS 6 32-bit. There is no CentOS 7 32-bit; it's 64-bit only.

#### Setup server and SSH:

1. Create a new VM for whatever OS version you're testing
2. Take a snapshot
3. Start your VM, login as root, and run one of these commands:
  * `ifup enp0s3` (for CentOS 7)
  * `ifup eth0` (for CentOS 6)
4. Get this branch's setup script using **cURL** (not `wget` anymore):
  * `curl -LO https://raw.githubusercontent.com/enterprisemediawiki/Meza1/centos7/setup.sh`
5. Run `bash setup.sh` as normal
6. Confirm you can login via SSH
7. Shutdown, take snapshot, restart

#### Run install script:

8. `curl -LO https://raw.githubusercontent.com/enterprisemediawiki/Meza1/centos7/client_files/install.sh`
9. `sudo bash install.sh` and make sure to select `centos7` when asked for git branch
10. When install complete do normal MediaWiki testing (create pages, search, use VE)
11. Create a new wiki (`create-wiki.sh`) and import an existing wiki or two (`import-wikis.sh`)
12. Import wikis
13. Confirm Elastic Search and other services continue to work after reboot.

#### Test in all environments:

1. Test on CentOS 7 (only 64-bit exists)
2. Test on CentOS 6 32-bit
3. Test on CentOS 6 64-bit

## Issues:

* Closes #163.
